Kernel/KResult.h

```diff
@@ -62,6 +62,18 @@ class alignas(T) KResultOr {
         other.m_error = KSuccess;
     }
 
+    KResultOr& operator=(KResultOr&& other)
+    {
+        m_is_error = other.m_is_error;
+        if (m_is_error)
+            m_error = other.m_error;
+        else
+            new (&m_storage) T(move(other.value()));
+        other.m_is_error = true;
+        other.m_error = KSuccess;
```

Comment:
We're forgetting to call the destructor on `value()` here. I don't think we should delay that until `~KResultOr()` unless we have a reason?

Also let's stick to the convention where a moved-from object looks as if it were just default-constructed.
A default-constructed `KResultOr` has `m_is_error = false` and `m_error = {}`.

Reply:
Hmm this was copy pasted from the move constructor, which leaves the moved-from object as is_error: yes, error = 0. It doesn't seem to make much sense to leave it in is_error = no, with a maybe constructed maybe not constructed T in the buffer to me? At least if is_error is true with an empty error the value is "defined". Though yeah, if 'this' has value it should probably be destroyed before we move construct a new T into the buffer.

Reply:
Also I'm not sure what you mean by "default constructed"? There isn't a default constructor for KResultOr, there's an error constructor, a T moving constructor, and a convert-U-to-T moving constructor. Seems reasonable to me that a moved-from object should be an error.

Reply:
Oh yeah duh. That's my bad, clearly `KResultOr` has no default constructor.
However, the way the move constructor and the new move assignment operator are written, we will neglect to invoke `~T()` on the moved-from object's `value()` after moving from it into our own `value()`.

Reply:
Oh yeah, because we tell the moved-from object that it's now an error and the ~T() won't get called ever. Wonder if any of the other containers have the same oversight :)